### DISCUSSION TITLE
Getting list of tickers

### DISCUSSION THREAD

**Discussion by jrbrodie77** [ranaroussi/yfinance#1512] [Show and tell]
Is there a way to get a list of current tickers?  I'd like to write something that walks through the entire nasdaq etc. and checks balance sheets, P/E etc. But as a starting point I need the ticker symbols.  Does yf or any API that you know of provide such a list?

**Comment by JetBlackHackerCat** [comment]
I don't know of an API that provides a list and yF doesn't support it currently but the NYSE does have an [FTP Share](ftp://ftp.nasdaqtrader.com/symboldirectory/) that has all the ticker symbols updated daily. You could write a script that accesses the FTP share and pulls them dynamically

**Comment by alvaroritter** [comment]
There is another package called yahoo_fin that does precisely what @MidnightTechnologies was describing. Here is the link to [yahoo_fin](https://theautomatic.net/yahoo_fin-documentation/). However, be aware that the methods are faulty at times. I already contacted the creator and sent him my suggestion for the corrected code and now created a pull request on his repository. 
Here is [my fork](https://github.com/alvaroritter/yahoo_fin) for the yahoo_fin package.

**Comment by earth-C137** [comment]
I've made a python3 script that does this: https://github.com/earth-C137/Stock-Symbol-List-Maker. You need to know how to code some to use it.  You have to manually download the file from the nasdaq ftp and write your file paths in the script.

**Comment by DimitriBolt** [comment]
> I've made a python3 script that does this: https://github.com/earth-C137/Stock-Symbol-List-Maker. You need to know how to code some to use it. You have to manually download the file from the nasdaq ftp and write your file paths in the script.

Thanks for the idea. I am going to code something similar.

**Comment by ValueRaider** [comment]
Copying over a message from @elamey

> Apologies, I did see this discussed in an earlier issue, I just can't find it now.
> 
> The question has been asked, where can we get lists of tickers and I thought I would share my method. I opened an acccount with Alpaca with $25. I don't trade there (yet), I just opened the account and parked the money there which gives access to thier api. Write a little python and you can get tickers and prices.
> 
> They have ~8300 securities available, most of them are in a format yfinanace will accept. There are some that yahoo won't accept. I personally just skip those. It takes about 90 minutes to download them because Alpaca throttles to 200 api hits per minute. Hope this helps.

**Comment by ValueRaider** [comment]
Another discussion: #1699
Hoping Github will let me merge these in future ...
